src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets

```diff
@@ -1,97 +1,31 @@
-    <AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets Condition="'$(AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets)' == ''">_GenerateProjectRestoreGraphAllFrameworks</AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets>
+    <PackageValidationBaselineName>$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$(PackageId)'))</PackageValidationBaselineName>
   </PropertyGroup>
 
+  <ItemGroup Condition="'$(PackageValidationBaselinePath)' == '' and '$(PackageValidationBaselineVersion)' != ''">
+    <PackageDownload Include="$(PackageValidationBaselineName)" Version="[$(PackageValidationBaselineVersion)]" />
+  </ItemGroup>
+
   <Target Name="RunPackageValidation"
           AfterTargets="Pack"
           Condition="$(IsPackable) == 'true'">
-    <PropertyGroup Condition="'$(EnablePackageBaselineValidation)' == 'true' and
-                              '$(PackageValidationBaselinePath)' == ''">
-      <PackageValidationBaselineName Condition="'$(PackageValidationBaselineName)' == ''">$(_PackageValidationBaselineName)</PackageValidationBaselineName>
-      <PackageValidationBaselineVersion Condition="'$(PackageValidationBaselineVersion)' == ''">$(_PackageValidationBaselineVersion)</PackageValidationBaselineVersion>
-      <PackageValidationBaselinePath Condition="'$(PackageValidationBaselineVersion)' != ''">$([MSBuild]::NormalizePath('$(NuGetPackageRoot)', '$(PackageValidationBaselineName.ToLower())', '$(PackageValidationBaselineVersion)', '$(PackageValidationBaselineName.ToLower()).$(PackageValidationBaselineVersion).nupkg'))</PackageValidationBaselinePath>
-      <EnablePackageBaselineValidation Condition="'$(PackageValidationBaselinePath)' == ''">false</EnablePackageBaselineValidation>
+
+    <PropertyGroup >
+      <PackageValidationBaselinePath Condition="'$(PackageValidationBaselinePath)' == '' and '$(PackageValidationBaselineVersion)' != ''">$([MSBuild]::NormalizePath('$(NuGetPackageRoot)', '$(PackageValidationBaselineName.ToLower())', '$(PackageValidationBaselineVersion)', '$(PackageValidationBaselineName.ToLower()).$(PackageValidationBaselineVersion).nupkg'))</PackageValidationBaselinePath>
     </PropertyGroup>
 
+    <Error Condition="!Exists('$(PackageValidationBaselinePath)')" Text="$(PackageValidationBaselinePath) does not exist. Please check the PackageValidationBaselinePath or PackageValidationBaselineVersion." />
+
     <!-- PackageTargetPath isn't exposed by NuGet: https://github.com/NuGet/Home/issues/6671. -->
     <Microsoft.DotNet.PackageValidation.ValidatePackage
       PackageTargetPath="$([MSBuild]::ValueOrDefault('$(PackageTargetPath)', '$(PackageOutputPath)$(PackageId).$(PackageVersion).nupkg'))"
       RuntimeGraph="$(RuntimeIdentifierGraphPath)"
       NoWarn="$(NoWarn)"
       RunApiCompat="$([MSBuild]::ValueOrDefault('$(RunApiCompat)', 'true'))"
-      BaselineValidation="$(EnablePackageBaselineValidation)"
```

Comment:
We currently set this property in dotnet/runtime to disable validation. How will this work without this property?

Reply:
we wont set PackageValidationBaselineVersion and PackageValidationBaselineFilePath for those projects.

Reply:
Sounds like a project author who creates a new package needs to unset the version i.e. this way: `<PackageValidationBaselineVersion />`. Do you believe that's the right user experience? Curious on your thoughts here .

Reply:
> Sounds like a project author who creates a new package needs to unset the version i.e. this way:

no he doesnt need to unset, because we are not setting it anywhere to a default value. we only set PackageValidationBaselineName to packageId.

Reply:
Not following. I thought we would set PackageValidationBaselineVersion to a default value in dotnet/runtime. How would you disable package baseline validation in a specific project?

Reply:
I was replying for a customer facing standalone project. A customer using this in his project and who doesnt want to run the baseline validation would not need to unset any variable.

yes for runtime repo, we default it 5.0.0. we are able to do that because our package versions are mostly synced.
 we will probably need to do some gymnastics to support where we dont want to run baseline validation.

But i think we should not add code to this customer feature which is specific to the runtime.

The way i imagine our customer scenarios are
- single standalone projects.
- multiple projects where probably the package versions (last released package versions) are not synced up.

Thats my thoughts :P

Reply:
> But i think we should not add code to this customer feature which is specific to the runtime.

You mean specific to repositories which use the same `PackageVersion` for their libraries. I'm sure that's not just dotnet/runtime.

> we will probably need to do some gymnastics to support where we dont want to run baseline validation.

With the current change, yes. That's why I'm asking if you think it's a good user experience in dotnet/runtime and similar repositories to unset the `PackageValidationBaselineVersion` or if `DisablePackageBaselineValidation` should be supported as well as an option.